src/_pytest/recwarn.py

```diff
@@ -322,10 +322,10 @@ def found_str():
             for w in self:
                 if not self.matches(w):
                     warnings.warn_explicit(
-                        str(w.message),
-                        w.message.__class__,  # type: ignore[arg-type]
-                        w.filename,
-                        w.lineno,
+                        message=w.message,  # type: ignore[arg-type]
```

Comment:
The docs say:

> message must be a string and category a subclass of [Warning](https://docs.python.org/3/library/exceptions.html#Warning) or message may be a [Warning](https://docs.python.org/3/library/exceptions.html#Warning) instance, in which case category will be ignored.

So I think this should be:

```suggestion
                        message=w,
```

No?

Reply:
Yes that makes sense!

Reply:
Hm, I'm having a bit of trouble with the mypy and the requirements for `warn_explicit()` (without just type ignoring or casting it). Would you mind taking a look?

Reply:
After trying it and reading the `warn_explicit` code, I think we should be good with just this:

```
                        message=w.message,
                        category=w.category,
```